x-pack/plugins/cases/server/connectors/cases/cases_connector_executor.ts

```diff
@@ -752,28 +746,44 @@ export class CasesConnectorExecutor {
     };
   }
 
-  private getCasesTitle(ruleName: string, oracleCounter: number) {
-    const suffix =
-      oracleCounter === INITIAL_ORACLE_RECORD_COUNTER
-        ? '(Auto-created)'
-        : `(${oracleCounter}) (Auto-created)`;
+  private getCasesTitle(
+    params: CasesConnectorRunParams,
+    grouping: GroupedAlerts['grouping'],
+    oracleCounter: number
+  ) {
+    const groupingDescription = Object.entries(grouping)
+      .map(([_, value]) => {
+        return convertValueToString(value);
+      })
+      .join(' & ');
 
-    const ruleNameTrimmed = ruleName.slice(0, MAX_TITLE_LENGTH - suffix.length - 1);
+    const suffix = `${
+      groupingDescription.length > 0 ? ` - Grouping by ${groupingDescription}` : ''
+    }${oracleCounter > INITIAL_ORACLE_RECORD_COUNTER ? ` (${oracleCounter})` : ''} (Auto-created)`;
 
-    return `${ruleNameTrimmed} ${suffix}`;
+    const ruleNameTrimmed = params.rule.name.slice(0, MAX_TITLE_LENGTH - suffix.length - 1);
+
+    return `${ruleNameTrimmed}${suffix}`;
   }
 
-  private getGroupingDescription(grouping: GroupedAlerts['grouping']) {
-    const flattenGrouping = getFlattenedObject(grouping);
+  private getCaseDescription(params: CasesConnectorRunParams, grouping: GroupedAlerts['grouping']) {
+    const ruleName = params.rule.ruleUrl
+      ? `[${params.rule.name}](${params.rule.ruleUrl})`
+      : params.rule.name;
 
-    return Object.entries(flattenGrouping)
-      .map(([key, value]) => {
-        const keyAsCodeBlock = `\`${key}\``;
-        const valueAsCodeBlock = `\`${convertValueToString(value)}\``;
+    const description = `This case was created by the Case action in ${ruleName}.`;
```

Comment:
What if we find a way to highlight/differentiate the rule name? Maybe by wrapping it in quotation marks or having it in italics?

I think currently it looks a bit messy:
<img width="1137" alt="Screenshot 2024-04-15 at 10 44 35" src="https://github.com/elastic/kibana/assets/1533137/1df36720-cd18-4d69-ba0a-457fb0a17f3d">

Reply:
Good point. If you see my screenshot in the description if you have set the `server.publicBaseUrl: 'https://localhost:5601'` in your `kibana.dev.yml` the name of the rule would be a URL. Quotation marks seem more suitable with links. Wdyt?

Reply:
Ah I like how it looks in the screenshot, imo that is ideal and we dont need the quotation marks.

Is there any scenario where no URL is rendered and we just have the name?

Reply:
Yes if the user is not set `server.publicBaseUrl` it will be shown as in your screenshot.